omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,236 @@
+			        .getPatientIdentifierTypeByUuid(identifierUuid);
+			autoGenerationOption.setLocation(location);
+			autoGenerationOption.setSource(identifierSource);
+			autoGenerationOption.setIdentifierType(patientIdentifierType);
+			autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+			autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+			
+		} else if (locationUuid != null && manualEntryEnabled != null && automaticGenerationEnabled != null
+		        && sourceUuid == null && identifierUuid != null) {
+			Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+			PatientIdentifierType patientIdentifierType = Context.getPatientService()
+			        .getPatientIdentifierTypeByUuid(identifierUuid);
+			autoGenerationOption.setLocation(location);
+			autoGenerationOption.setIdentifierType(patientIdentifierType);
+			autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+			autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+			
+		} else {
+			throw new ResourceDoesNotSupportOperationException(
+			        "Creating an autoGenerationOption with missing parameters is not supported");
+		}
+		
+		return (SimpleObject) ConversionUtil.convertToRepresentation(save(autoGenerationOption), Representation.DEFAULT);
+	}
+	
+	@Override
+	public SimpleObject update(String uuid, SimpleObject postBody, RequestContext context) throws ResponseException {
+		String manualEntryEnabled = postBody.get("manualEntryEnabled").toString();
+		String automaticGenerationEnabled = postBody.get("automaticGenerationEnabled").toString();
+		String sourceUuid = postBody.get("source").toString();
+		String locationUuid = postBody.get("location").toString();
+		PatientIdentifierType patientIdentifierType = Context.getPatientService().getPatientIdentifierTypeByUuid(uuid);
+		AutoGenerationOption autoGenerationOption = Context.getService(IdentifierSourceService.class)
+		        .getAutoGenerationOption(patientIdentifierType);
+		
+		if (manualEntryEnabled != null) {
+			autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+		}
+		
+		if (automaticGenerationEnabled != null) {
+			autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+		}
+		
+		if (sourceUuid != null) {
+			IdentifierSource identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid);
+			autoGenerationOption.setSource(identifierSource);
+		}
+		
+		if (locationUuid != null) {
+			Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+			autoGenerationOption.setLocation(location);
+		} else {
+			throw new ResourceDoesNotSupportOperationException(
+			        "updating an autoGenerationOption with missing parameters is not supported");
```

Comment:
What if i just want to update only one, automaticGenerationEnabled? Would you force me to update all the others?